### PR TITLE
feat(frontend): expose backend allow_signing function

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

We need to expose the new `allow_signing` function at the API level for #2632.

# Changes

- Adding `allowSigning` to backend canister and API

### CODE REVIEW

**Comment by DenysKarmazynDFINITY** [comment]
Gonna come back after the PR is updated with the code from `main` 🙂

**Comment by peterpeterparker** [comment]
@DenysKarmazynDFINITY updated. go.

File: src/frontend/src/lib/canisters/backend.canister.ts [hunk 1]
```diff
@@ -161,4 +162,10 @@ export class BackendCanister extends Canister<BackendService> {
 
 		return response.Ok;
 	};
+
+	allowSigning = async (): Promise<AllowSigningResponse> => {
+		const { allow_signing } = this.caller({ certified: true });
+
+		return allow_signing();
```
Comment (DenysKarmazynDFINITY) [inline src/frontend/src/lib/canisters/backend.canister.ts:169]: Shall we handle possible response errors here, similar to other functions? Or in this particular case you need the whole `AllowSigningResponse`?

Comment (peterpeterparker) [reply src/frontend/src/lib/canisters/backend.canister.ts:169]: Yes but I'll do it in another PR given that I provided this PR before you introduced the pattern.

Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/canisters/backend.canister.ts:169]: Got it

**Review by DenysKarmazynDFINITY — APPROVED** [review]
LGTM, thanks!

**Pull request merged**